### ISSUE TITLE
How to use input equirectangular images for training?

### ISSUE THREAD

**Issue by HungNgoCT** [nerfstudio-project/nerfstudio#1252]
Hi,

I saw in a PR that nerfstudio support equirectangular images as this link: 
https://github.com/nerfstudio-project/nerfstudio/pull/1024

Can anyone guide me that how to use equirectagular images for training?
I understood that I will use ns-process-data for for images and pose generation, then use ns-train for training. I checked ns-process-data but don't know how to use it with input equirectangular images.

Highly appreciate if anyone can help.

Hung
![2_000004](https://user-images.githubusercontent.com/110378166/213141124-fa4d679e-06f3-4961-87c8-6fc311a8790c.jpg)

**Comment by machenmusik** [comment]
For clarity - I see https://github.com/nerfstudio-project/nerfstudio/pull/966 which claims it adds equirectangular image support, but it sounds as though there is no ns-process-data method that uses that yet - is that right?

**Comment by mgupta70** [comment]
I have been experimenting with 360 videos. I split them into multiple overlapping perspective images and register those in colmap. Registration was successful but the 3D reconstruction with Instant NGP is not good. Although, The dense 3D reconstruction in Colmap is much better. I tried the same set of photos in RealityCapture and 3d reconstruction is comparatively of a very high quality. 

Can you think of a reason why it could happen?


![colmap](https://user-images.githubusercontent.com/110718346/219115698-08c4feaf-a659-46b3-851f-cb6f19de74d3.png)
![instantNGP](https://user-images.githubusercontent.com/110718346/219115706-a66834a7-3809-4b96-8b97-eccc9eaf6fd4.png)
![reality_capture](https://user-images.githubusercontent.com/110718346/219115710-ef6a26a5-ea24-41ed-8cb7-a062270becda.png)

**Comment by mgupta70** [comment]
I did not try with nerfacto. Will try it and post it here. But any guesses @tancik as to why this could be happening ?

**Comment by tancik** [comment]
Maybe you don't have enough images. I'd also like to see a viewpoint from inside the room.

**Comment by mgupta70** [comment]
I have attached an equirectangular image and two of the perspective images from it. I have 12 equirectangular images and 120 perspective images. 

![9](https://user-images.githubusercontent.com/110718346/219125609-697c28bc-f42e-4819-b2d1-f7fbd81f4fc5.jpg)
![9_71](https://user-images.githubusercontent.com/110718346/219125797-70f92d67-8523-4dd0-afd2-784d0582e9e8.jpg)
![9_80](https://user-images.githubusercontent.com/110718346/219125807-1ded422c-6150-4dff-be7c-9c3a05c7a5af.jpg)

**Comment by tancik** [comment]
Can you post a screenshot of a nerf reconstruction from inside the room. Right away though, I can tell you that the helmet in the view will cause issues. It will likely lead to lots of floating geometry around the locations where the images were captured

**Comment by mgupta70** [comment]
It does create a lot of floating geometry. I was wondering why and now what you said makes sense.
I am attaching a small video from inside the reconstructed scene
https://user-images.githubusercontent.com/110718346/219147344-a7fca84a-5aab-4d42-b2fe-b34bd4c2e919.mp4

**Comment by tancik** [comment]
It looks like you aren't using nerfstudio.

**Comment by mgupta70** [comment]
Will do it 🤞

**Comment by mgupta70** [comment]
I used nerfacto and split images into 8 perspective images (from meshlab) such that helmet is not present. I am attaching a equirectangular image and 2 of its perspective images. The reconstruction was not good again. I am attaching snapshots of it however, path of cameras and camera poses look accurate because I captured the data.
![raw360_ASU_Multipurpose_Arena_1631973616167](https://user-images.git
![frame_00071](https://user-images.githubusercontent.com/110718346/219465588-be0aa862-c5b5-49e6-bca3-f49a623d11db.jpg)
![frame_00067](https://user-images.githubusercontent.com/110718346/219465592-396d816a-9072-4eec-870e-d3566d165475.jpg)
hubusercontent.com/110718346/219465511-43892f9d-c1f0-4586-a064-ccf911c27074.jpg)
![im2](https://user-images.githubusercontent.com/110718346/219465873-d46734ab-aa36-48cc-9dd9-9939d15a5c8e.png)
![im1](https://user-images.githubusercontent.com/110718346/219465875-7d3fa03c-39c9-4ce1-98c5-df91b53f4d01.png)

**Comment by mgupta70** [comment]
yes, I did it with` ns-process-data` which then ran Colmap 3.8. I cannot understand why is this happening?

I have 2 queries in this regard: 
1. Can it be related to the scale of cube in which we render? (Like aabb_scale in NGP)?
2. Is it a good idea to remove those images which have less information. I mean For ex: if an image only has plain white wall in background and only a small portion in that image is occupied by another object. 

Thank you @tancik for replying

**Comment by tancik** [comment]
1. The nerfacto model autoscales the scene so you don't need to worry about aabb_scale.
2. Including that data is fine.

Can you send me your processed data.

**Comment by mgupta70** [comment]
How can I send it to you? Email ?

**Comment by tancik** [comment]
I tested the data, it also doesn't work for me, Im not sure why. We will be releasing some 360 processing tools soon, they might help with this data.

**Comment by mgupta70** [comment]
oh :(
Thanks for looking. I will be eagerly waiting for the updates. Meanwhile, I will play around with other parameters like `Sampler`  and `matching method` to be `exhaustive`

**Comment by tancik** [comment]
https://user-images.githubusercontent.com/3310961/219978377-f0136119-94b8-4a80-843a-0140df51204e.mp4

There was a bug introduced in nerfstudio 0.1.17
A patch will be coming soon. Alternatively you can use 0.1.16 in the meantime.

**Comment by mgupta70** [comment]
Woah!!! This looks good. 
 Thanks a lot. I will shift to 0.1.16.

**Comment by tancik** [comment]
v0.1.18 is out, use that instead.
